### DISCUSSION TITLE
Connecting to OSCam Server failed

### DISCUSSION THREAD

**Discussion by BaltasarParreira** [Barracuda09/SATPI#125] [Q&A] [answered]
I Know there is already a few explanations about using OSCAM here, but I setup everything using enter to save and so on, but still can't connect to may local OSCAM where SATPI is running.
Always only getting:
src/decrypt/dvbapi/Client.cpp:229] Connecting to OSCam Server failed

Is this still working with the latest OSCAM and SATPI?

Also can you detail for what are these parameters in the OSCAM configuration page:
- OSCam Aadapter offset
- Rewrite PMT

Thanks.

**Comment by Barracuda09** [comment]
Hi @BaltasarParreira 

First of all thanks for your feedback.

Well I don't use the cutting edge version of OSxxx (_I use version R11272_). I know the API has changed over time and probably not working in combination with SatPI at the moment.

Regards,
Marc

**Comment by BaltasarParreira** [comment]
Ok, going to install that version and see if that's the problem.
Anyway any specific oscam.conf parameters apart the usual:
[dvbapi]
enabled                       = 1
au                            = 1
pmt_mode                      = 2
listen_port                   = 9999
delayer                       = 60
read_sdt                      = 1
write_sdt_prov                = 1

You answer so quickly that I was editing the previous post, but I ask again anyway regarding those parameters I put there.  ; )

Thanks.

**Comment by BaltasarParreira** [comment]
Sorry still not working, I'm running now the OSCam r11272 but I'm getting the same error.

My SATPI version is: 1.5.356~g1f7aa3e Enigma

Did you made any changes that could solve this problem after this version?

Thanks.

**Comment by BaltasarParreira** [comment]
Answering myself regarding SATPI version, I just compile the latest SATPI, now running: 1.5.409~g45892cc

But same problem, can't connect to OSCAM.
Don't really know what I can do more so waiting for your feed back please.

Thanks.

**Comment by Barracuda09** [comment]
I have these as API
[dvbapi]
enabled  = 1
au           = 1
pmt_mode = 4
listen_port  = 15012
user            = TEST
read_sdt     = 2
boxtype     = pc

**Comment by BaltasarParreira** [comment]
Ok and did you create the user "TEST"?

**Comment by Barracuda09** [comment]
Yes I did

**Comment by BaltasarParreira** [comment]
Ok, still nothing.

So my scenario is, a Gigablue UHD UE 4K receiver running OSCAM and SATPI inside.
Is this ok for the SATPI configuration?
![image](https://user-images.githubusercontent.com/1650425/110857641-02a32280-82b1-11eb-82af-2593c5c8c528.png)

**Comment by Barracuda09** [comment]
I have this:

![image](https://user-images.githubusercontent.com/6542445/110858103-f7082980-82b9-11eb-9c98-398d08ae62bc.png)

**Comment by BaltasarParreira** [comment]
Nothing... really strange since I have everything identical.
This is my oscam.version giving by the interface:

```
Unix starttime: 1615499023
Starttime:      11.03.2021 21:43:43
Version:        oscam-1.20-unstable_svn-r11272
Compiler:       arm-newnigma2-linux-gnueabihf
Box type:       gigablue (generic)
PID:            2701
TempDir:        /tmp/.oscam
ConfigDir:      /etc/tuxbox/config/oscam-stable/
WebifPort:      83

Web interface support:                   yes
LiveLog support:                         yes
jQuery support intern:                   yes
Touch interface support:                 no
SSL support:                             no
DVB API support:                         yes
DVB API with AZBOX support:              no
DVB API with MCA support:                no
DVB API with COOLAPI support:            no
DVB API with COOLAPI2 support:           no
DVB API with STAPI support:              no
DVB API with STAPI5 support:             no
DVB API read-sdt charsets:               yes
Irdeto guessing:                         yes
Anti-cascading support:                  no
Debug mode:                              yes
Monitor:                                 yes
Loadbalancing support:                   yes
Cache exchange support:                  no
CW Cycle Check support:                  no
LCD support:                             no
LED support:                             no
Clockfix with realtime clock:            no
IPv6 support:                            no

camd 3.3x:                               no
camd 3.5 UDP:                            yes
camd 3.5 TCP:                            yes
newcamd:                                 yes
CCcam:                                   yes
CCcam share:                             yes
gbox:                                    yes
radegast:                                no
scam:                                    yes
serial:                                  no
constant CW:                             no
Pandora:                                 no
ghttp:                                   no

Reader support:                          yes

Nagra:                                   yes
Irdeto:                                  yes
Conax:                                   yes
Cryptoworks:                             yes
Seca:                                    yes
Viaccess:                                yes
NDS Videoguard:                          yes
DRE Crypt:                               yes
TONGFANG:                                yes
Bulcrypt:                                yes
Griffin:                                 yes
DGCrypt:                                 yes

cardreader_phoenix:                      yes
cardreader_drecas:                       no
cardreader_internal_azbox:               no
cardreader_internal_coolapi:             no
cardreader_internal_coolapi2:            no
cardreader_internal_sci:                 yes
cardreader_sc8in1:                       no
cardreader_mp35:                         no
cardreader_smargo:                       no
cardreader_pcsc:                         no
cardreader_smartreader:                  no
cardreader_db2com:                       no
cardreader_stapi:                        no
cardreader_stapi5:                       no
cardreader_stinger:                      yes
```

Do you have any other protocol or something that is missing in mine?

**Comment by Barracuda09** [comment]
You did restart OSCam I assume.

**Comment by BaltasarParreira** [comment]
Yes I did.
I'm starting SATPI only like this:
satpi --http-path /usr/share/satpi/web

Do you use any other parameter?

**Comment by Barracuda09** [comment]
I will convert this to a discussion

**Comment by Barracuda09** [comment]
I use this on my Vu+:

satpi --http-path /usr/satpi/web --app-data-path /usr/satpi --iface-name eth0

**Reply by BaltasarParreira** [reply]
Yupiiiii, I managed to connect now, I just realize what you meant with the restart OSCAM.

After SATPI start I have to restart OSCAM so it works, even the OSCAM config didn't change, but it's the only way that SATPI connects to him.

Perfect, thanks !!!
Going now to test newer OSCAM version to see if still works and I let you know.

**Reply by BaltasarParreira** [reply]
Ok found the initial problem, it works with the latest OSCAM:
![image](https://user-images.githubusercontent.com/1650425/110864074-85c87680-82b9-11eb-836d-859d059b2fed.png)

The problem is I have to have on oscam.conf:
boxtype                       = pc

If I use:
boxtype                       = dreambox

SATPI stoped beeing able to conect, so this only works if the receiver is not decoding at the same time, right?

**Reply by Barracuda09** [reply]
It should be possible if you set _Adapter Offset_ to lets say 10. Then it should not interfere.

**Reply by BaltasarParreira** [reply]
The problem is on OSCAM side, if I use boxtype = dreambox the listen port apart being save on the file, is not working and the next time I go to the DVB Api configuration the listen port value is empty.

**Comment by BaltasarParreira** [comment, marked as answer]
Solved everything in a crazy way.  ; ) 

I installed OSCAM on a Rapberry Pi just to receive the connection from SATPI, them from the Pi OSCAM I create a reader for cs378x protocol to connect to the Gigablue receiver and everything works and all channels decode at the same time.
Like this I can watch a channel on the Gigablue while on the Phone or Pad someone can watch other channel trough SATPI.

To solve and avoid this and since OSCAM used as other boxtype apart the pc  type and if not too much effort you should start thinking in use something like the minisatip tool approach using the connection straight to the camd.socket file.

Now that this is solved my second goal is to have a way of creating a channellist.m3u with all the channels at least from one satellite so VLC on a normal PC could get it, instead the free default channels that shows, since doesn't have a way of doing scan like some other clients, I think.
Any idea on this?

Thanks for all the help and support !!!

**Comment by jack8282** [comment]
I'm trying to run satpi with oscam but the field is empty. I use open atv 6.5  oscam_1.20-rev-11272-Pli-4.0-emu VU+ solo2. I set it up oscam dvb api port.  but the field is empty in satpi.    start-stop-daemon -S -x /usr/bin/satpi -p /var/lock/SatPI.lock -- --http-path /usr/share/satpi/web $SATPI_OPTS > /dev/null    That's how it starts 1.5.465~g9baec6e Enigma

**Comment by Barracuda09** [comment]
Hi @jack8282 

You don't see this in SatPI?
https://github.com/Barracuda09/SATPI/discussions/125#discussioncomment-471712

Then most likely SatPI is not compiled (build) with dvbapi support 

Edit:
You can take a look here:
https://github.com/Barracuda09/SATPI/wiki/Setup-OSCam-DVBAPI-Protocol

**Comment by jack8282** [comment]
No I don't see the writing fields just the oscam field. There are no lines to write

**Comment by Barracuda09** [comment]
Hi @jack8282 

then it is not build with **`make LIBDVBCSA=yes`**

**Comment by jack8282** [comment]
I will write to you in the mail make mips  new bin make LIBDVBCSA=yes .  Thank you so much  I really like how it works satpi

**Reply by jack8282** [reply]
dvbapi (client: SatPI 1.5.472~g4b20e24 Enigm, protocol version: 2)     OSCam: 1.20_svn Build: r11701 Compiler: x86_64-linux-gnu  connects to the computer  thanks a lot of work.   oscam X64  was launched elsewhere and tied up

**Comment by jack8282** [comment]
Can it simultaneously decode 2-3 channels from 1 transponder in this way dvbapi (client: SatPI

**Comment by Barracuda09** [comment]
Hi @BaltasarParreira 

Please checkout latest commits, I have done some fixes for OSCam problems.
